### ISSUE TITLE
Certain widgets occasionally having issues scrolling

### ISSUE THREAD

**Issue by dustinkerstein** [zobnin/aio#38]
I don't have perfect replication steps, but it seems to happen a few times a day with the Todoist Task list widget. 

**To Reproduce**
1. Add the Todoist Task list widget to the homescreen (position doesn't seem to matter nor does its size)
2. Ensure there are enough tasks in the list the result in scrolling being necessary based on the size of the widget
3. Verify that scrolling works
4. [TBD]
5. Once scrolling is broken, it can be fixed by holding the AIO search and hitting the Edit button, and then immediately exiting the Edit mode

**Expected Behavior**
Scrolling should always function if there are contents to scroll. Note that I have not seen this on other widgets which allow scrolling (ie. Google Calendar). 

**Smartphone**
Device: Pixel XL
Android version: LineageOS 18.1
AIO version: 4.5.2

https://user-images.githubusercontent.com/7659306/187085825-00e8471f-1ebd-4a31-9096-b070ce17c99e.mov

**Comment by zobnin** [comment]
Have you tried using this widget in other launchers?

**Comment by dustinkerstein** [comment]
Yeah, I've used this widget for the past couple of years with the Pixel XL's default launcher and have never seen that behavior. One potentially interesting note is that sometimes when scrolling that widget in other launchers (tested on LineageOS 18.1 Trebuchet and Pixel default launcher), the row level Todoist tasks below the fold, that haven't been shown, will occasionally show a spinner for each item while they load. This doesn't happen often, and I've yet to be able to force replicate it in those other launchers, but the timing of that loading / spinner behavior in the other launchers feels suspiciously like the timing / frequency of when scrolling breaks in AIO. So maybe there's a connection between when the widget needs to make a call / receive an event... But that's a total guess. But I do know that so far in my testing, I've yet to see that individual row level spinner in the Todoist widget when using AIO.

**Comment by dustinkerstein** [comment]
Quick update - I finally did see those row level spinners in the Todoist widget when using AIO, so I'm not sure that's related. However, I do believe I found a way to replicate, even though it's not the only way I've seen this issue. 

1. Add the Todoist widget in AIO
2. Ensure there are enough tasks in the current view to require scrolling
3. Restart the phone and wait for the widget to load (this can take a few minutes and during my replication I've continued using the phone - ie. opening other apps)
4. Note that scrolling is broken until going into the AIO edit homescreen mode

**Comment by dustinkerstein** [comment]
Another quick update - When scrolling breaks, rotating the phone fixes it (from portrait to landscape, and remains working when rotated back to portrait).

**Comment by zobnin** [comment]
Widget reloaded on rotation so it is logical. Other than that, I don't know what the problem is yet.

**Comment by joaodiasafonso** [comment]
Hi.
I have this problem too in my Business Calendar widget. It always happens ("freezes" vertical scrolling) at device startup. What I am already used to do is, after a device restart, open AIO Android definitions and "force restart". After that, it keeps working until next phone reboot.
